Library/Homebrew/formula.rb

```diff
@@ -476,6 +476,29 @@ def compiler_failures
     active_spec.compiler_failures
   end
 
+  # The best spec we can upgrade the {Formula} to so it can satisfy dependency.
+  def best_upgrade_spec_sym
```

Comment:
> > I think HEAD should be updated to stable when we can detect it's too outdated i.e. when stable have changed since we last installed HEAD
>
> It's tough; I'm not sure how many people will want to always follow HEAD vs. just upgrade to stable.

When I install a HEAD formula, it's almost always because I need a specific thing that hasn't been released yet — so when a new stable version is released, I'd be happy to be switched to tracking stable.

However, there's also the use-case where you always want to use HEAD because you want to help find issues before they're released or whatever.

So I think maybe both options should be available somehow — both use-cases seem like something worth covering. I'd make the "update HEAD to stable" use-case the default, but maybe there could be some kind of `pin`-like functionality for forcing upgrades to a specific formula to track HEAD.